docs/examples/auto-instrumentation/README.rst

```diff
@@ -0,0 +1,142 @@
+
+    # server_uninstrumented.py
+    @app.route("/server_request")
+    def server_request():
+        print(request.args.get("param"))
+        return "served"
+
+Preparation
+-----------
+
+This example will be executed in a separate virtual environment:
+
+.. code-block:: sh
+
+    $ mkdir auto_instrumentation
+    $ virtualenv auto_instrumentation
+    $ source auto_instrumentation/bin/activate
+
+Installation
+------------
+
+.. code-block:: sh
+
+    $ pip install opentelemetry-api
+    $ pip install opentelemetry-sdk
+    $ pip install opentelemetry-auto-instrumentation
+    $ pip install ext/opentelemetry-ext-flask
+    $ pip install flask
+    $ pip install requests
+
+Execution
+---------
+
+Execution of the manually instrumented server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is done in 2 separate consoles, one to run each of the scripts that
+make up this example:
+
+.. code-block:: sh
+
+    $ source auto_instrumentation/bin/activate
+    $ python opentelemetry-python/opentelemetry-auto-instrumentation/example/server_instrumented.py
+
+.. code-block:: sh
+
+    $ source auto_instrumentation/bin/activate
+    $ python opentelemetry-python/opentelemetry-auto-instrumentation/example/client.py testing
+
+The execution of ``server_instrumented.py`` should return an output
+similar to:
+
+.. code-block:: sh
+
+    Hello, testing!
+    Span(name="serv_request", context=SpanContext(trace_id=0x9c0e0ce8f7b7dbb51d1d6e744a4dad49, span_id=0xd1ba3ec4c76a0d7f, trace_state={}), kind=SpanKind.INTERNAL, parent=None, start_time=2020-03-19T00:06:31.275719Z, end_time=2020-03-19T00:06:31.275920Z)
+    127.0.0.1 - - [18/Mar/2020 18:06:31] "GET /serv_request?helloStr=Hello%2C+testing%21 HTTP/1.1" 200 -
+
+Execution of an automatically instrumented server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now, kill the execution of ``server_instrumented.py`` with ``ctrl + c``
+and run this instead:
+
+.. code-block:: sh
+
+    $ opentelemetry-auto-instrumentation python opentelemetry-python/opentelemetry-auto-instrumentation/example/server_uninstrumented.py
+
+In the console where you previously executed ``client.py``, run again
+this again:
+
+.. code-block:: sh
+
+    $ python opentelemetry-python/opentelemetry-auto-instrumentation/example/client.py testing
+
+The execution of ``server_uninstrumented.py`` should return an output
+similar to:
+
+.. code-block:: sh
+
+    Hello, testing!
+    Span(name="serv_request", context=SpanContext(trace_id=0xf26b28b5243e48f5f96bfc753f95f3f0, span_id=0xbeb179a095d087ed, trace_state={}), kind=SpanKind.SERVER, parent=<opentelemetry.trace.DefaultSpan object at 0x7f1a20a54908>, start_time=2020-03-19T00:24:18.828561Z, end_time=2020-03-19T00:24:18.845127Z)
+    127.0.0.1 - - [18/Mar/2020 18:24:18] "GET /serv_request?helloStr=Hello%2C+testing%21 HTTP/1.1" 200 -
+
+As you can see, both outputs are equivalent since the automatic
+instrumentation does what the manual instrumentation does too.
+
```

Comment:
Should we add more advanced uses cases here? For example [passing in configuration arguments in](https://github.com/open-telemetry/opentelemetry-python/pull/579) `instrument`.